src/models/PrimaryAuth.js

```
@@ -110,7 +110,7 @@ export default BaseLoginModel.extend({
       } else {
         primaryAuthPromise = this.doTransaction(function(transaction) {
           return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
-        });
+        }, true);
```

Does true mean "yes, hide spinner"?

---

take a look at `doPrimaryAuth` function..

---

Found in `doTransaction`, the second arg is `rethrow`. So it _was_ swallowing the error, but passing `true` will rethrow the error and then trigger 'removeLoading' on line 126. Cool.

---

Yup.. You got it.

---

Nice I didn't know this either 🙂